src/components/calcite-dropdown-group/calcite-dropdown-group.tsx

```diff
@@ -0,0 +1,152 @@
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  @Event() calciteDropdownItemHasChanged: EventEmitter;
+  @Event() registerCalciteDropdownGroup: EventEmitter;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  componentDidLoad() {
+    this.groupPosition = this.getGroupPosition();
+    this.items = this.sortItems(this.items);
+    this.registerCalciteDropdownGroup.emit({
+      items: this.items,
+      position: this.groupPosition
+    });
+  }
+
+  render() {
+    const theme = getElementTheme(this.el);
+    const dropdownState = {
+      requestedDropdownGroup: this.requestedDropdownGroup,
+      requestedDropdownItem: this.requestedDropdownItem
+    };
+
+    const grouptitle = this.grouptitle ? (
+      <span class="dropdown-title">{this.grouptitle}</span>
+    ) : null;
+
+    return (
+      <Host theme={theme} id={this.dropdownGroupId}>
+        {grouptitle}
+        <DropdownInterface.Provider state={dropdownState}>
+          <slot />
+        </DropdownInterface.Provider>
+      </Host>
+    );
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("calciteDropdownItemSelected") updateActiveItemOnChange(
+    event: CustomEvent
+  ) {
+    this.requestedDropdownGroup = event.detail.requestedDropdownGroup;
+    this.requestedDropdownItem = event.detail.requestedDropdownItem;
+
+    this.calciteDropdownItemHasChanged.emit({
+      requestedDropdownGroup: this.requestedDropdownGroup,
+      requestedDropdownItem: this.requestedDropdownItem
+    });
+  }
+
+  @Listen("registerCalciteDropdownItem") registerCalciteDropdownItem(
+    e: CustomEvent
+  ) {
+    const item = {
+      item: e.detail.item as HTMLCalciteDropdownItemElement,
+      position: e.detail.position
+    };
+    this.items.push(item);
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private State/Props
+  //
+  //--------------------------------------------------------------------------
+
+  /** created list of dropdown items */
+  @State() private items = [];
+
+  /** @internal */
+  @State() private groupPosition: number;
+
+  /** unique id for dropdown group */
+  /** @internal */
+  private dropdownGroupId = `calcite-dropdown-group-${guid()}`;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private getGroupPosition() {
+    const groupPosition = Array.prototype.indexOf.call(
+      this.el.parentElement.querySelectorAll("calcite-dropdown-group"),
```

Comment:
I think you could do this with spread like:

```
[...this.el.parentElement.querySelectorAll("calcite-dropdown-group")].indexOf(this.el)
```

But not totally sure as I haven't done it within the stencil tooling